app/assets/javascripts/application.js

```diff
@@ -42,3 +42,42 @@ function show_notification_blood(msg){
 }
 
 
+function prepare_to_initialize(q) {
+  $(document).ready(function() {
+   google.maps.event.addDomListener(window, 'load',function () { initialize(q);});
+});
+}
+
+
+ function initialize(x) {
+
+  var input = document.getElementById(x);
+  var options = {
+    componentRestrictions: {country: "eg"}
+  };
+
+  var autocomplete = new google.maps.places.Autocomplete(input,options);
+
+ if (x == "searchTextField") {
```

Comment:
We are looking for the id (x), I think using a class will remove all the duplication